### PR TITLE
EPD-415 Add prompt template manager

### PR DESCRIPTION
**Pull request by nicolewhite**
Screenshot 2023-11-13 at 4 02 05 PM

Screenshot 2023-11-13 at 4 02 22 PM

Screenshot 2023-11-13 at 4 02 30 PM

### CODE REVIEW

**Review by adamnolte — APPROVED** [review]

**Comment by nicolewhite** [comment]
@adamnolte will you take a look at the README and let me know if it makes sense to you

File: src/prompts-cli/index.ts [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+// We autogenerate these based on the templates directory.
+// See src/prompts/autogenerated.ts.
+const AUTOGENERATED_INTERFACE_NAME = '__AutogeneratedTemplatePathToParameters';
+const AUTOGENERATED_VARIABLE_NAME = '__autogeneratedTemplatePathToTemplate';
+
+const autogenerationConfigs: AutogenerationConfig[] = [
+  {
+    symbolName: AUTOGENERATED_INTERFACE_NAME,
+    symbolAppearanceBeforeAutogeneration: `interface ${AUTOGENERATED_INTERFACE_NAME} {\n}`,
+    filesToModify: ['../prompts/index.d.ts', '../prompts/index.d.mts'],
```
Comment (adamnolte) [inline src/prompts-cli/index.ts:34]:
```suggestion
    filesToModify: ['../prompts/index.d.ts', '../prompts/index.d.mjs'],
```
?

Comment (adamnolte) [reply src/prompts-cli/index.ts:34]: or .mjs doesn't seem right either haven't seen mts before

Comment (adamnolte) [reply src/prompts-cli/index.ts:34]: or actually is it just `index.d.ts` here?

File: src/prompts/README.md [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+```
+
+Placeholders can have arbitrary whitespace, so this is also valid:
+
+```txt
+My name is {{name}}.
+My age is {{      age       }}
+```
+
+> **_HINT:_** Placeholders aren't always replaced with constants. They can also be replaced with the value of another rendered template.
```
Comment (adamnolte) [inline src/prompts/README.md:229]: can we add an example of this? Like language or point to that.

**Review by adamnolte — APPROVED** [review]
Looks good! I know currently we are keeping all docs on the docs page and readmes here minimal. Are you planning to move the readme to the docs then?

File: src/prompts/README.md [hunk 3]
```diff
@@ -0,0 +1,294 @@
...
+autoblocks prompts generate
+```
+
+Note: You'll likely want to add this to your `package.json` scripts so that it is easier to run:
+
+`package.json`:
+
+```json
+"scripts": {
+  "ab-prompts-gen": "autoblocks prompts generate"
```
Comment (adamnolte) [inline src/prompts/README.md:243]: I think people should also be able to do `npx autoblocks prompts generate` will have to test though and we can update

File: src/prompts-cli/index.ts (hunk 1 above)
Comment (nicolewhite) [reply src/prompts-cli/index.ts:34]: No we need to do both type files here (this is the interface, not the value)

Comment (adamnolte) [reply src/prompts-cli/index.ts:34]: Oh i see, i didn't recognize the .mts extension. Just looked at the dist folder though and see it.

**Comment by nicolewhite** [comment]
Ya I can move it , was just easier to draft here

File: src/prompts/README.md (hunk 3 above)
Comment (nicolewhite) [reply src/prompts/README.md:243]: I think it won't work because `autoblocks` isn't the name of the package

File: src/prompts/README.md (hunk 2 above)
Comment (nicolewhite) [reply src/prompts/README.md:229]: Ya I can update when I move to the docs

File: src/prompts/README.md (hunk 3 above)
Comment (adamnolte) [reply src/prompts/README.md:243]: ya we might have to include the package name

**Review by adamnolte — APPROVED** [review]

**Pull request merged**